File: tests/test-minecraft/src/main/java/pize/tests/minecraft/game/gui/text/formatting/Style.java
package pize.tests.minecraft.game.gui.text.formatting;

import pize.graphics.util.color.IColor;
import pize.graphics.util.color.ImmutableColor;

public class Style{

    public static final Style DEFAULT = new Style(TextFormatting.WHITE.color());


    public final ImmutableColor color;
    public final boolean bold;
    public final boolean italic;
    public final boolean underline;
    public final boolean strikethrough;
    public final boolean obfuscated;
    public final boolean obfuscated_numbers;

    public Style(IColor color, boolean bold, boolean italic, boolean underline, boolean strikethrough, boolean obfuscated, boolean obfuscated_numbers){
        this.color = new ImmutableColor(color);
        this.bold = bold;
        this.italic = italic;
        this.underline = underline;
        this.strikethrough = strikethrough;
        this.obfuscated = obfuscated;
        this.obfuscated_numbers = obfuscated_numbers;
    }

    public Style(IColor color, boolean... style){
        this.color = new ImmutableColor(color);
        this.bold = style.length > 0 && style[0];
        this.italic = style.length > 1 && style[1];
        this.underline = style.length > 2 && style[2];
        this.strikethrough = style.length > 3 && style[3];
        this.obfuscated = style.length > 4 && style[4];
        this.obfuscated_numbers = style.length > 5 && style[5];
    }

}

File: tests/test-minecraft/src/main/kotlin/pize/tests/minecraft/game/gui/text/formatting/Style.kt
package pize.tests.minecraft.game.gui.text.formatting

import pize.graphics.util.color.IColor
import pize.graphics.util.color.ImmutableColor

class Style {
    val color: ImmutableColor
    val bold: Boolean
    val italic: Boolean
    val underline: Boolean
    val strikethrough: Boolean
    val obfuscated: Boolean
    val obfuscated_numbers: Boolean

    constructor(
        color: IColor?,
        bold: Boolean,
        italic: Boolean,
        underline: Boolean,
        strikethrough: Boolean,
        obfuscated: Boolean,
        obfuscated_numbers: Boolean
    ) {
        this.color = ImmutableColor(color!!)
        this.bold = bold
        this.italic = italic
        this.underline = underline
        this.strikethrough = strikethrough
        this.obfuscated = obfuscated
        this.obfuscated_numbers = obfuscated_numbers
    }

    constructor(color: IColor?, vararg style: Boolean) {
        this.color = ImmutableColor(color!!)
        bold = style.size > 0 && style[0]
        italic = style.size > 1 && style[1]
        underline = style.size > 2 && style[2]
        strikethrough = style.size > 3 && style[3]
        obfuscated = style.size > 4 && style[4]
        obfuscated_numbers = style.size > 5 && style[5]
    }

    companion object {
        val DEFAULT = Style(TextFormatting.WHITE.color())
    }
}
